src/group_theory/nilpotent.lean

```diff
@@ -328,6 +336,78 @@ begin
     use [lower_central_series G, lower_central_series_is_descending_central_series, h] },
 end
 
+section classical
+
+open_locale classical
+
+variables [hG : is_nilpotent G]
+include hG
+
+/-- The nilpotency class of a nilpotent group is the small natural `n` such that
+the `n`'th term of the upper central series is `G`. -/
+noncomputable def group.nilpotency_class : ℕ :=
+nat.find (is_nilpotent.nilpotent G)
+
+/-- Equivalently, the nilpotency class is the smallest `n` for which any ascending
```

Comment:
These docstrings are used to show mouse tooltips when people hover over a declaration name. Also outside the context of this file. So it would be good if the docstring is a stand-alone sentence that makes sense outside the context of this file.
```suggestion
/-- The nilpotency class is the smallest `n` for which any ascending
```

Reply:
Then maybe the following, to avoid misleading anyone in believing that this is “the” definition?
```suggestion
/-- The nilpotency class is equal to the smallest `n` for which any ascending
```